include/civetweb.h

```diff
@@ -334,6 +334,15 @@ mg_set_websocket_handler(struct mg_context *ctx,
                          mg_websocket_close_handler close_handler,
                          void *cbdata);
 
+/* mg_set_auth_handler
+
+   Sets or removes a URI mapping for an authorization handler.
+   This function works similar to mg_set_request_handler - see there. */
+CIVETWEB_API void mg_set_auth_handler(struct mg_context *ctx,
+                                      const char *uri,
+                                      mg_request_handler handler,
```

Comment:
I originally did this. However, I ended up reusing the 'handler' member of 'mg_request_handler_info' as it greatly simplified the changes to the 'mg_set_request_handler_type' function. If I split out the handler to it's own type, I will probably change (in about 4 or 5 places):

if (is_websocket_handler) {
/* handle websockets */
} else {
/* handle request and auth */
}

to

switch (handler_type) {
case REQUEST_HANDLER:
/* handle request*/
case WEBSOCKET_HANDLER:
/* handle websocket*/
case AUTH_HANDLER:
/* handle auth*/
}